### DISCUSSION TITLE
GitHub Copilot : "access has ended." And ""rate limit exceeded for plan FREE_EDUCATIONAL" on Student Developer Pack

### DISCUSSION THREAD

**Discussion by fahim9778** [community/community#43893] [Copilot Conversations] [answered]
_I live in Bangladesh, South Asia, hence all times mentioned in this post are on _**GMT+6**_._

# TL;DR:

1. GitHub Copilot stopped working on 8th January at around 11 AM, showing no more suggestions.
2.  Visiting the [Copilot page](https://github.com/settings/copilot) on my GitHub profile says _(in a pale yellow text box)_

> "Thank you for participating in the GitHub Copilot Technical Preview. Your extended access has ended. "

but it contradicts on [Billing and Payment Page](https://github.com/settings/billing) which shows "`

> "You are eligible to use GitHub Copilot for free."

along with the prior message.

3. Somewhere between 12 - 1 AM on 9th January, Copilot seemed to work but after giving 1-2 consecutive suggestions, it gets stuck with an error message 

> "Copilot is encountering temporary issues(click for details)"

In the error log of VS Code, it prints out 
`[ERROR] [fetch] [2023-01-09T17:56:36.650Z] Unhandled status from server:,429,{"error":{"message":"rate limit exceeded","internal_message":"rate limit exceeded for plan FREE_EDUCATIONAL","type":"client_error"}}`

4. The problems still persist at the time of this writing (_11 January 2023, 12:12 AM_), and users around the globe are facing the same problems. ([Students](https://github.com/orgs/community/discussions/43841), [Faculties](https://github.com/orgs/community/discussions/43673), [Open Source Maintainers](https://github.com/orgs/community/discussions/43673#discussioncomment-4638523))

# Community Discussions that reveal these issues:

- Existing Verified Students Developer Pack Owners (Got automated Emails of Renewal on 11th December 2022 although there  [#43634](https://github.com/orgs/community/discussions/43634)
- Some got a renewal, but they are getting `rate limit exceeded` [error messages ](https://github.com/orgs/community/discussions/43634#discussioncomment-4644223) and [here](https://github.com/orgs/community/discussions/43634#discussioncomment-4647338)
- Surprisingly enough, **[some of the annual subscribers](https://github.com/orgs/community/discussions/43865)** are getting the `"rate limit exceeded for plan FREE_EDUCATIONAL"` error message, even though they have active subscriptions as stated.

# My Subscriptions and Billing Timeline:

1. I subscribed to Copilot on June 22, 2022, at 1:40 AM via my Student Developer Pack. It says trial on this![Screenshot 2023-01-11 at 00-51-19 Welcome to GitHub Copilot! - fahim9778@gmail com - Gmail](https://user-images.githubusercontent.com/26436023/211637016-1b88540c-e30d-4531-9447-3a616bee9e83.png) but [I have got free access](https://www.facebook.com/photo/?fbid=5500044823390957&set=p.5500044823390957&opaqueCursor=AbqK62UXBh1ahpOve7dA46sbLtJ3zuAeUxJ2YrbhVKqN3b5-vvBzqnhgmimYoleZfapf7gHmePEh5UHSSzqlbe6zNfygIRb8Eh1ZfuPb85_mp55X_Ja4IpLbuddxq0SBajXnE1gwJE7ScfJar_XI2knJrR5huttwUIoE3My0XRH0DMLJV6Asc6Ajkles_9q96CEd0A32bGRpR1FFCAwd6S20CoWSlNNMCG4xiYetQSlUT3PrBRJ4PwN4oQMYYGrxwk5BxflDtBzSGn5vZx5swjF3EKGIJlzkcq3btAktnhImlckJMejSxZS86fniCX7CJDFF57hRWKIT1-zuvAsGpsri6NZZpzTEaL9CrPkA2leCm3qbbWyvdYM6ZvN7GUzNd3E6JGWAmTA8yEaPVdQYsAZ2fRx3PyZYuuU7Sitf7Mv375cj1WMdjzk2F4H7YembI12N05IVOpyGNKN0nvoSSwPxJWRNk2WTNQS8RR0X0MRrLhxZK58YpepvYj3BHosJFl6WbTWaDMfclGvU71VXLbn6ufcAQ6hyNB3VLHzabh1OdwwvjwAJZWToi_TtjFmIKEHgyFSGdwPZuIJARK4u_v2h) as per Student Developer Pack Offers 
![288998543_5500044920057614_5588631287068595800_n](https://user-images.githubusercontent.com/26436023/211638143-3b89d9c5-8e54-4eac-b779-788b4e65d321.jpg)

2. On December 11th, 2022, I got this email saying my Copilot will end on 9th January 2023 and most of the troubled users have received this email _(even though they have initially subscribed on different dates)_
![image](https://user-images.githubusercontent.com/26436023/211645303-1bd386c8-f44f-49a0-83de-9c5f1661fef7.png)

3. I still have the **Verified Student Developer Pack** and can't reverify. 
![firefox_idd7phgfcc](https://user-images.githubusercontent.com/26436023/211638828-2ce69ace-f823-49a8-844f-693556dbca24.png)

4. My GitHub Billing Page says 
![firefox_gMPbiWWVuy](https://user-images.githubusercontent.com/26436023/211639189-ffb02c9f-e641-4f4c-b16a-2ef25714c064.png)

5. Although [GitHub Copilot official Doc](https://docs.github.com/en/billing/managing-billing-for-github-copilot/about-billing-for-github-copilot#pricing-for-github-copilot-for-individuals) says 
![image](https://user-images.githubusercontent.com/26436023/211641341-0a08102e-5999-4cd6-bbe7-d137b4ca1e47.png)

### It's super stressful and disappointing that **I can't switch plans as I am eligible for Free Use** and at the same time I have a `rate limit exceeded for plan FREE_EDUCATIONAL`.

# Consfusions and Conclusion:

- **Is GitHub changing Plans for Free Tier Users?** Imposing a `rate limit` on the OpenAI Codex API calls? 
- - Then **_let us know in a clear announcement via Email_** or Post. Also, let the users change their plans. **There is no meaning in keeping them on Free Tier and suffering as the sudden break of Copilot Completions is super frustrating.**
- If there are internal Bugs and fixes, let some @github-staff report and support the users in clear communications.
- - I have posted on Support Channel, [Twitter ](https://twitter.com/fahim9778/status/1612666659671805953), and also GitHub Next's support discord profile, but it seems every staff member has gone dark. Strange! 

I really hope someone responsible comes into play and lets us know quickly what's going on and what'll happen. Most of us have projects, assignments, and other stuff depending on Copilot and they need to be resolved quickly **_(very quickly!)_**

**_Thanks for reading._**

**Comment by Akash1134** [comment, marked as answer]
Hi @fahim9778 ,

Some details and update on this issue:

This week, we've had to take down some of our Copilot resources for maintenance. While we did our best to source extra capacity elsewhere, we've been forced to limit some free users for the period of time where one of our busiest models is out of commission.

We expect that this will cease to impact your use of Copilot after the 16th of January. In the meantime, you will have less frequent suggestions, but you should still be getting suggestions - just with rate limits in between. You may see these errors specifically if you look in the output pane or logs for your IDE.

Thanks for your patience and apologies for the inconvenience caused.

**Reply by fahim9778** [reply]
@Akash1134
Thank you for the update. I really wish it back to normal. Facing too much difficulties right now.  😣

**Comment by pyronaur** [comment]
I tried upgrading, but since I already have a free plan associated with my account, I can't upgrade to a paid plan, so looks like people with a free plan are stuck on a rate-limit for now with no way out.
